daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/Decode.scala

```diff
@@ -58,11 +58,13 @@ object Decode {
   ): (PackageId, Ast.Package) =
     assertRight(decodeArchive(archive, onlySerializableDataDefs))
 
-  def decodeInfoPackage(archive: DamlLf.Archive): Either[Error, PackageInfo] =
+  def decodeInfoPackage(
+      archive: DamlLf.Archive
+  ): Either[Error, ((PackageId, Ast.Package), PackageInfo)] =
```

Comment:
Why not a 3-tuple instead of a ((A,B),C)?

Reply:
I considered, but I thought they are better grouped logically like this as the PackageInfo is based on (PackageId, Ast.Package)

Reply:
Looking at how PackageInfo is constructed in practice, I think (PackageId, Ast.Package) <-> PackageInfo could be isomorphic, and exposing the former two directly as getters on the PackageInfo could have been enough. But that would mean changing the signature of PackageInfo into accepting a tuple instead of a map, so avoided